pkg/cmd/target.go

```diff
@@ -739,6 +769,57 @@ func gardenWrapper(targetReader TargetReader, targetWriter TargetWriter, configR
 	return nil
 }
 
+func serverWrapper(reader ConfigReader, serverName string, kubeconfigReader KubeconfigReader) {
+	config := reader.ReadConfig(pathGardenConfig)
+
+	if findIP(serverName) != "" {
+		for _, garden := range config.GardenClusters {
+			kc := garden.KubeConfig
+			svr := getServerValueFromKubeconfig(kc, kubeconfigReader)
+			addrIP, err := net.LookupIP(removeHTTPSandPort(svr))
+			checkError(err)
+			if addrIP[0].String() == removeHTTPSandPort(serverName) {
+				pgarden = garden.Name
+			}
+		}
+	} else {
+		for _, garden := range config.GardenClusters {
+			kc := garden.KubeConfig
+			svr := getServerValueFromKubeconfig(kc, kubeconfigReader)
+			if svr == serverName {
+				pgarden = garden.Name
+			}
+		}
+	}
+
+	if pgarden == "" {
+		fmt.Println("a garden could not be matched to the provided server address:", serverName)
+		os.Exit(2)
+	}
+
+}
+
+func findIP(input string) string {
+	numBlock := "(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])"
+	regexPattern := numBlock + "\\." + numBlock + "\\." + numBlock + "\\." + numBlock
+
+	regEx := regexp.MustCompile(regexPattern)
+	return regEx.FindString(input)
+}
+
+func checkServerName(input string) bool {
+	if strings.HasPrefix(input, "https://") && strings.HasSuffix(input, ":443") {
+		return true
+	}
+	return false
+}
```

Comment:
```suggestion
func isValidUri(input string) bool {
	_, err := url.ParseRequestURI(input)
	return err == nil
}
```

Reply:
this suggestion is already included in https://github.com/neo-liang-sap/gardenctl/pull/1